src/lib.rs

```diff
@@ -603,6 +603,40 @@ impl Uuid {
         Ok(uuid)
     }
 
+    /// Creates a `Uuid` using the supplied bytes.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// use uuid::Uuid;
+    /// use uuid::UuidBytes;
+    ///
+    /// let bytes:UuidBytes = [70, 235, 208, 238, 14, 109, 67, 201, 185, 13, 204, 195, 90, 145, 63, 62];
+    ///
+    /// let uuid = Uuid::from_uuid_bytes(&bytes);
+    /// let uuid = uuid.hyphenated().to_string();
+    ///
+    /// let expected_uuid = String::from("46ebd0ee-0e6d-43c9-b90d-ccc35a913f3e");
+    ///
+    /// assert_eq!(expected_uuid, uuid);
+    /// ```
+    ///
+    /// An incorrect number of bytes:
+    ///
+    /// ```compile_fail
+    /// use uuid::Uuid;
+    /// use uuid::UuidBytes;
+    ///
+    /// let bytes:UuidBytes = [4, 54, 67, 12, 43, 2, 98, 76]; // doesn't compile
+    ///
+    /// let uuid = Uuid::from_uuid_bytes(&bytes);
+    /// ```
+    pub fn from_uuid_bytes(b: &UuidBytes) -> Uuid {
```

Comment:
Since we just take a copy of these bytes, and `UuidBytes` is sized, maybe this could just take an owned `UuidBytes` instead of a `&UuidBytes` here? The clone will be really cheap, but it could be clearer that we take ownership of the raw bytes.

What do you think?

Reply:
`from_bytes()` also accepts a reference. So this keeps it consistent with that. Also I guess it is better to change the return type to a `Result` right?

Reply:
Hmm, `from_bytes` takes a reference because it needs to, otherwise it's unsized. But `UuidBytes` isn't unsized, so we can take it by value if we want, since we need it by value.

I don't think we'd need to return a `Result`, because this always succeeds.

Reply:
Yes, I deliberately didn't return a `Result`, because this always succeeds, as KodrAus noted. One motivation for this function (for me) is that I want to be able to create const Uuid's (if/when const functions come out of Rust nightly). I believe (thought I'm not positive) that if this returns a Result, this function wouldn't be able to become a const function that could generate a const Uuid.

I did have a reference in the signature to mirror `from_bytes()`, as noted by Dylan-DPC. However, I didn't have a strong motivational reason for this, other than matching the interface. I think KodrAus makes a good point that it might be better than the consumer of this function should be aware that a copy of the bytes are going to happen. As such, I'd be happy to change the interface to take an owned UuidBytes, and require the caller to perform the `.clone` if they want the copy.

I can make that change tomorrow morning (PST) and send the results. I can include the nightly rustfmt change at that time, too.

Reply:
Yeah i agree with it. Cool